### ISSUE TITLE
Type hints

### ISSUE THREAD

**Issue by thijsvandien** [SethMMorton/natsort#132]
The library does not yet provide type hints for use with Mypy for example. It would be great to get rid of `# type: ignore` and enjoy the additional safety.

**Comment by SethMMorton** [comment]
`natsort` is pretty flexible with types - I am not sure how much help type hints would provide:

```python

def natsort(seq: Iterable[Any], key: Callable[[Any], Any], reverse: bool, alg: int): -> List[Any]:
    ...
```

Having said that - I am not against it. Is this something you are willing to submitting a PR to implement?

**Comment by thijsvandien** [comment]
Not in the short term, but knowing it would be welcomed, in time I might.

**Comment by domdfcoding** [comment]
If anyone wants a starting point I made [domdfcoding/natsort-stubs](https://github.com/domdfcoding/natsort-stubs) for my own use. Feel free to use them.

I think I made with with mypy's stubgen and added what else I could infer by hand. The only issue I ran into was with the `ns` object. I faked it as an Enum as mypy wasn't keen on the namedtuple.

**Comment by SethMMorton** [comment]
Closed with #138.

**Comment by thijsvandien** [comment]
Congratulations on getting this done so swiftly!
